test/bench_httpclient.c

```diff
@@ -163,7 +163,9 @@ launch_request(void)
 	evutil_gettimeofday(&ri->started, NULL);
 
 	b = bufferevent_socket_new(base, sock, BEV_OPT_CLOSE_ON_FREE);
-
+	if (b == NULL) {
+		return -1;
```

Comment:
Should close the fd as well, and also let's improve the caller to print which request failed